src/shared/linux/bpf.h

```diff
@@ -0,0 +1,673 @@
+/* Copyright (c) 2011-2014 PLUMgrid, http://plumgrid.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of version 2 of the GNU General Public
+ * License as published by the Free Software Foundation.
+ */
```

Comment:
Likewise, this file cannot be present in the systemd sources. If we use a public header provided by the system, that's different. But including it in our sources is not possible w/o bumping our own license to full GPL.

Reply:
naah... we don't have to bump anything. As mentioned above I don#t think this is a real problem at all. But if indeed the header's copyright would matter, then we could still license our stuff as LGPL, though effectively it would be downgraded to GPL when you actually run it. But GPL vs LGPL only matters during runtime really, not in what we include in our tarball. 

Or in other words, as long as we tell people that if they turn on the BPF stuff PID 1 will be effectively GPL it doesn't really matter at all...

I mean, quite frankly I wouldn't even have a problem at all with systemd being GPL rather than LGPL. The only reason we care so far, is that libsystemd should be linkable by non-free code, and we want the ability to copy/paste around our own code between libsystemd and PID 1 and other components freely without constantly thinking about the license...

IANAL, of course, but i don't think this is really a problem, at all. This is not code after all, just APIs...

Reply:
(and btw, some components of systemd are GPL rather than LPGL anyway, src/basic/ioprio.h, src/basic/securebits.h, and some stuff of udev iirc. That's why we ship LICENSE.GPL2 in the repo after all)